spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/UniProtEntryCommentsConverter.java

```diff
@@ -89,10 +95,11 @@ void convertCommentToDocument(List<Comment> comments, UniProtDocument document)
 
             switch (comment.getCommentType()) {
                 case CATALYTIC_ACTIVITY:
-                    convertCatalyticActivity((CatalyticActivityComment) comment, document);
+                    convertCatalyticActivity(
```

Comment:
This switch-case and convertX methods can ideally be converted into a more manageable code using a pattern, ex: Strategy pattern (something to do later)